superset/db_engine_specs/trino.py

```diff
@@ -193,34 +201,32 @@ def execute_with_cursor(
         in another thread and invoke `handle_cursor` to poll for the query ID
         to appear on the cursor in parallel.
         """
+        # Fetch the query ID before hand, since it might fail inside the thread due to
+        # how the SQLAlchemy session is handled.
+        query_id = query.id
```

Comment:
Ah, this explains what sqlite was complaining about then; I never did get to the bottom of that.